src/test/java/net/sf/jabref/importer/fileformat/InspecImportTest.java

```diff
@@ -0,0 +1,109 @@
+package net.sf.jabref.importer.fileformat;
+
+import static org.junit.Assert.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.BibtexEntryTypes;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InspecImportTest {
+
+    private InspecImporter inspecImp;
+
+    @Before
+    public void setUp() throws Exception {
+        Globals.prefs = JabRefPreferences.getInstance();
+        this.inspecImp = new InspecImporter();
+    }
+
+    @Test
+    public void testIsRecognizedFormatAccept() throws IOException {
+        List<String> testList = Arrays.asList("InspecImportTest.txt", "InspecImportTest2.txt");
+        for (String str : testList) {
+            try (InputStream inStream = InspecImportTest.class.getResourceAsStream(str)) {
+                assertTrue(inspecImp.isRecognizedFormat(inStream));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatReject() throws IOException {
+        List<String> testList = Arrays.asList("CopacImporterTest1.txt", "CopacImporterTest2.txt",
+                "IEEEImport1.txt", "IsiImporterTest1.isi", "IsiImporterTestInspec.isi", "IsiImporterTestWOS.isi",
+                "IsiImporterTestMedline.isi", "RisImporterTest1.ris", "InspecImportTestFalse.txt");
+
+        for (String str : testList) {
+            try (InputStream inStream = InspecImportTest.class.getResourceAsStream(str)) {
+                assertFalse(inspecImp.isRecognizedFormat(inStream));
+            }
+        }
+    }
+
+    @Test
+    public void testCompleteBibtexEntryOnJournalPaperImport() throws IOException {
+        try (InputStream inStream = InspecImportTest.class.getResourceAsStream("InspecImportTest2.txt")) {
+            List<BibEntry> entries = inspecImp.importEntries(inStream, new OutputPrinterToNull());
+            assertEquals(1, entries.size());
+            BibEntry entry = entries.get(0);
+
+            assertEquals("The SIS project : software reuse with a natural language approach", entry.getField("title"));
+            assertEquals("Prechelt, Lutz", entry.getField("author"));
+            assertEquals("1992", entry.getField("year"));
+            assertEquals("Abstrakt", entry.getField("abstract"));
+            assertEquals("key", entry.getField("keywords"));
+            assertEquals("10000", entry.getField("journal"));
+            assertEquals("20", entry.getField("pages"));
+            assertEquals("19", entry.getField("volume"));
+            assertEquals(BibtexEntryTypes.ARTICLE.getName().toLowerCase(), entry.getType());
```

Comment:
You can have the input/output file pairs inline, if you like. That is not an issue. 

Well, you could cheat but why would you? For me, a test is much easier to understand if I can reduce it to a simple input/output system. If I have to rely on a canonical format, this is not a unit test anymore, I admit. But testing only the input/output pairs you have an acceptance test which could be shown to a user as well and the user would understand this. And the higher a test, the more stable it will be. A unit test in this regard may be modified if you change any of the methods of the bibentry. 

To me, the input/output is much more maintainable, as this could be easily applied to all importers. Hence, we could just add triples of (importer, input, output) and this could be used to test the importer. Even users could create test cases and a single test class could implement this as a data driven tests.